### PR TITLE
Migrate to Java 11

### PR DESCRIPTION
**Pull request by h-arlt**
- update project dependencies and switch to Java11
- relocate Maven projects to `com.xceptance` group
- get rid of XCMStarter and use NinjaJetty instead

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -3,20 +3,50 @@
...
+        <module>xcmailr-load-test-suite</module>
     </modules>
 
+    <properties>
+        <!-- pin encoding to UTF-8 -->
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.target>11</maven.compiler.target>
+         <copyright>Copyright (c) ${project.inceptionYear} ${project.organization.name}</copyright>
```
Comment (jowerner) [inline pom.xml:30]:
```suggestion
         <copyright>Copyright (c) ${project.inceptionYear}-2023 ${project.organization.name}</copyright>
```

File: xcmailr-webapp/bin/export.sh
Comment (jowerner) [inline xcmailr-webapp/bin/export.sh:1]: What about making this script similar to `import.sh` with respect to the usage of `realpath` and `exec`?

File: xcmailr-webapp/conf/application.conf.template
Comment (jowerner) [inline xcmailr-webapp/conf/application.conf.template:1]: IMO this file should be identical to `src/main/java/conf/application.conf`, but isn't.

File: xcmailr-webapp/license.md
Comment (jowerner) [inline xcmailr-webapp/license.md:1]: Please leave this file in the root directory of the repo. The license applies to all sub modules alike.

File: xcmailr-webapp/src/main/java/conf/application.conf [hunk 2]
```diff
@@ -162,25 +164,20 @@ ebean.ddl.run=false
...
-ebean.datasource.maxConnections=25
+ebean.datasource.maxConnections=50
 ebean.datasource.isolationlevel=read_committed
 ebean.datasource.heartbeatsql=select 1
 
-# uncomment the following line to enable stacktrace capturing used to detect connection leaks
-#eban.datasource.captureStackTrace=true
-%test.ebean.datasource.captureStackTrace=true
-%dev.ebean.datasource.captureStackTrace=true
-
```
Comment (jowerner) [inline xcmailr-webapp/src/main/java/conf/application.conf:173]: The capture stack trace properties seem useful to me. What about keeping them, maybe comment out?

File: xcmailr-webapp/src/test/java/services/MessageListenerDeliverTest.java [hunk 3]
```diff
@@ -38,7 +38,8 @@ public class MessageListenerDeliverTest extends NinjaTest
     @Test
     public void testDeliveryPreconditions() throws Exception
     {
-        final NinjaProperties ninjaProperties = spy(new NinjaPropertiesImpl(NinjaMode.test));
+        final NinjaProperties props = NinjaPropertiesImpl.builder().withMode(NinjaMode.test).build();
+        final NinjaProperties ninjaProperties = spy(props);
```
Comment (jowerner) [inline xcmailr-webapp/src/test/java/services/MessageListenerDeliverTest.java:42]: Calling `spy(props)` is not needed here. Use the opportunity and remove that?

File: xcmailr-webapp/src/main/java/services/CheckDBForMailAddressDuplicates.java [hunk 4]
```diff
@@ -38,8 +38,8 @@ private void validateUserAddresses() throws Throwable
     {
         final HashSet<String> mailAddresses = new HashSet<>();
         final HashMap<Long, String> idToMailAddress = new HashMap<>();
-        final PagingList<User> users = Ebean.find(User.class).select("id, mail").findPagingList(1_000);
-        for (final User user : users.getAsList())
+        final PagedList<User> users = Ebean.find(User.class).select("id, mail").setMaxRows(1_000).findPagedList();
+        for (final User user : users.getList())
```
Comment (jowerner) [inline xcmailr-webapp/src/main/java/services/CheckDBForMailAddressDuplicates.java:42]: Please check if this construct really finds and processes all users or just the first 1,000.

File: xcmailr-webapp/src/main/java/services/CheckDBForMailAddressDuplicates.java [hunk 5]
```diff
@@ -68,8 +68,8 @@ private void validateMailboxAddresses() throws Throwable
     {
         final HashSet<String> mailAddresses = new HashSet<>();
         final HashMap<Long, String> idToMailAddress = new HashMap<>();
-        final PagingList<MBox> boxes = Ebean.find(MBox.class).select("id, address, domain").findPagingList(1_000);
-        for (final MBox box : boxes.getAsList())
+        final PagedList<MBox> boxes = Ebean.find(MBox.class).select("id, address, domain").setMaxRows(1_000).findPagedList();
+        for (final MBox box : boxes.getList())
```
Comment (jowerner) [inline xcmailr-webapp/src/main/java/services/CheckDBForMailAddressDuplicates.java:72]: Please check if this construct really finds and processes all mailboxes or just the first 1,000.

File: xcmailr-resources/conf/keystore.jks
Comment (jowerner) [inline xcmailr-resources/conf/keystore.jks:1]: Do we want to keep this file as a ready-to-run dev keystore in `xcmailr-webapp/conf`?

**Review by jowerner — CHANGES_REQUESTED** [review]
See conversations on the Conversation tab.

File: pom.xml [hunk 6]
```diff
@@ -3,20 +3,50 @@
...
     <artifactId>xcmailr</artifactId>
-    <groupId>org.xcmailr</groupId>
+    <version>3.0.0</version>
     <packaging>pom</packaging>
+    
     <name>XCMailr</name>
-    <version>2.2.2</version>
-
+    <url>https://github.com/Xceptance/XCMailr</url>
+    <inceptionYear>2020</inceptionYear>
```
[the remark this answers is not in the document]
Comment (h-arlt) [reply pom.xml]: Done.

File: pom.xml (hunk 1 above)
Comment (h-arlt) [reply pom.xml]: Done.

File: xcmailr-webapp/bin/shell.sh [hunk 7]
```diff
@@ -32,7 +32,7 @@ if [ -n "$DB_PASS" ]; then
   ARGS="$ARGS -pass $DB_PASS"
 fi
 
-if [ $# -lt 1 ]; then
+if [ $# -gt 0 ]; then
```
[the remark this answers is not in the document]
Comment (h-arlt) [reply xcmailr-webapp/bin/shell.sh:35]: Done.

File: xcmailr-webapp/conf/application.conf.template
Comment (h-arlt) [reply xcmailr-webapp/conf/application.conf.template:1]: Files should be identical now (except of application.secret setting).

File: xcmailr-webapp/src/main/java/conf/application.conf (hunk 2 above)
Comment (h-arlt) [reply xcmailr-webapp/src/main/java/conf/application.conf:173]: Done.

File: xcmailr-webapp/src/main/resources/logging.properties
[the remark this answers is not in the document]
Comment (h-arlt) [reply xcmailr-webapp/src/main/resources/logging.properties:1]: Added `jul-to-slf4j' to dependencies and initialize SLF4j bridge upon application start; adjusted logback configuration files to include proper logger definitions for subethamail and JavaMail.

Just for clarification: subethamail does not use JUL at all but its dependency JavaMail does.

File: xcmailr-webapp/src/main/java/services/CheckDBForMailAddressDuplicates.java (hunk 4 above)
Comment (h-arlt) [reply xcmailr-webapp/src/main/java/services/CheckDBForMailAddressDuplicates.java]: Changed to iterator approach (recommended for queries w/ large result sets).

File: xcmailr-webapp/src/main/java/services/CheckDBForMailAddressDuplicates.java (hunk 5 above)
Comment (h-arlt) [reply xcmailr-webapp/src/main/java/services/CheckDBForMailAddressDuplicates.java]: Changed to iterator approach (recommended for queries w/ large result sets).

File: xcmailr-resources/conf/keystore.jks
Comment (h-arlt) [reply xcmailr-resources/conf/keystore.jks:1]: File 'keystore_dev.jks'  is the ready-to-run dev keystore configured for Ninja Maven plugin in POM.

**Comment by jowerner** [comment]
What about updating the license headers in `xcmailr-webapp`, `xcmailr-client`, and `xcmailr-load-test-suite` via `mvn license:format` in each subdirectory?

File: xcmailr-webapp/pom.xml [hunk 8]
```diff
@@ -157,51 +168,41 @@
             </resource>
         </resources>
```
Comment (jowerner) [inline xcmailr-webapp/pom.xml:169]: I would like to keep the `version.ftl.html file` together with all the other template files so there will be no confusion where this file is located. The following `resources` definition makes this possible.

```
        <resources>
            <resource>
                <directory>src/main/java</directory>
                <excludes>
                    <exclude>**/*.java</exclude>
                    <exclude>views/layout/version.ftl.html</exclude>
                </excludes>
            </resource>
            <resource>
                <directory>src/main/java</directory>
                <filtering>true</filtering>
                <includes>
                    <include>views/layout/version.ftl.html</include>
                </includes>
            </resource>
            <resource>
                <directory>src/main/resources</directory>
            </resource>
        </resources>
```

WDYT? Would of course require the file to be moved from `src\main\resources` to `src\main\java\views\layout`.

**Review by jowerner — CHANGES_REQUESTED** [review]
Just two more little things ...

File: xcmailr-webapp/pom.xml (hunk 8 above)
Comment (h-arlt) [reply xcmailr-webapp/pom.xml:167]: Moved all resource files to 'src/main/resources' and 'src/test/resources' respectively following Maven conventions.

Comment (jowerner) [reply xcmailr-webapp/pom.xml:167]: All right. In this case we can also get rid of this block:

```
            <resource>
                <directory>src/main/java</directory>
                <excludes>
                    <exclude>**/*.java</exclude>
                </excludes>
            </resource>
```

Comment (h-arlt) [reply xcmailr-webapp/pom.xml:161]: Narf... Nice catch! Done.

**Review by jowerner — APPROVED** [review]

**Pull request merged**